### DISCUSSION TITLE
Add fuzzy search to the search bar

### DISCUSSION THREAD

**Discussion by revolter** [espanso/espanso#1163] [Feature requests and Ideas]
**Is your feature request related to a problem? Please describe.**

I have some matches with long names for them to be explicit and clear.

**Describe the solution you'd like**

I'd like to be able to type only some letters of the words composing the trigger to find it/

**Describe alternatives you've considered**

\-

**Additional context**

Related to https://github.com/federico-terzi/espanso/issues/420#issuecomment-787133857.

**Comment by federico-terzi** [comment]
Hey @revolter,

Thank you for the feedback! If anyone wants to give it a try, we should simply implement a new module here: https://github.com/federico-terzi/espanso/blob/master/espanso-modulo/src/search/algorithm.rs

Cheers :)

**Reply by TobeMagic** [reply]
hello, I see this yet complete, Can I have a try now?

**Reply by smeech** [reply]
> hello, I see this yet complete, Can I have a try now?

It's a longstanding working feature of the program. I don't understand your question, I'm afraid. Can I help?

**Comment by alexn-s** [comment]
**love espanso, especially with the search bar.** (last time i checked espanso i did not have one)

+1 for this feature.
i noticed i am in need of fuzzy search/ search settings to modify the behavior for ones needs
(e.g. i dont want to use space to search separate words in the label property)

**Comment by davideimola** [comment]
I would like to make a try!

**Comment by CharlesARoy** [comment]
I use a command-line cheatsheet tool called [Navi](https://github.com/denisidoro/navi) which has implemented this well using Fzf. I can't speak to its implementation though.

**Comment by AucaCoyan** [comment]
I like this!
